### PR TITLE
Ignore not found error when rotating log file

### PR DESCRIPTION
**Pull request by sunxiaoguang**
Do not return not found error when log file was deleted before rotate.

Signed-off-by: Xiaoguang Sun <sunxiaoguang@zhihu.com>

## What have you changed? (mandatory)
Do not return error when log file is deleted before rotating. Although I doubt if we really need to return any error at here. It is definitely better to be conservative and only skip std::io::ErrorKind::NotFound at this time. 

Please explain **IN DETAIL** what the changes are in this PR and why they are needed:
- Summarize your change (mandatory)
TiKV is going to panic during rotating if current log file is deleted. This PR changes log rotating behavior and ignore NotFound error kind. 

Please **NOTE** that:
- Do not assume reviewers understand the original issue

## What are the type of the changes? (mandatory)
- Misc (other changes)
NA

## How has this PR been tested? (mandatory)
Unit test has been updated to validate the change.

## Does this PR affect documentation (docs) or release note? (mandatory)
No

- If there is a document change, please file a PR in ([docs]
No
- If this PR should be mentioned in the release note, please update the [release note]
No

## Does this PR affect tidb-ansible update? (mandatory)
No

If there is a configuration or metrics change, please file a PR in [tidb-ansible]
No

## Refer to a related PR or issue link (optional)
NA

## Benchmark result if necessary (optional)
NA

## Add a few positive/negative examples (optional)
NA

### CODE REVIEW

File: components/tikv_util/src/logger/file_log.rs [hunk 1]
```diff
@@ -67,7 +67,11 @@ impl RotatingFileLogger {
 
         // Note: renaming files while they're open only works on Linux and macOS.
         let new_path = rotation_file_path_with_timestamp(&self.file_path, &Utc::now());
-        fs::rename(&self.file_path, new_path)?;
+        match fs::rename(&self.file_path, new_path) {
```
Comment (siddontang) [inline components/tikv_util/src/logger/file_log.rs]: it is better to use a failpoint to add a test here.

Comment (sunxiaoguang) [reply components/tikv_util/src/logger/file_log.rs]: Sounds like a good idea. Let me put one here.

Comment (sunxiaoguang) [reply components/tikv_util/src/logger/file_log.rs]: @siddontang failpoint 'file_log_rename' added.

**Comment by sunxiaoguang** [comment]
Test failed mysteriously at not related part. How to trigger a new run of it please?

**Comment by sunxiaoguang** [comment]
> Test failed mysteriously at not related part. How to trigger a new run of it please?

Merging from master did the trick.

**Pull request merged**